packages/headless/src/controllers/field-suggestions/category-facet/headless-category-field-suggestions.ts

```diff
@@ -13,16 +74,135 @@ import {
-  extends CategoryFacetSearch,
-    Subscribable {}
+export interface CategoryFieldSuggestions extends Subscribable {
+  /**
+   * Requests field suggestions based on a query.
+   *
+   * @param text - The query to search.
+   */
+  updateText(text: string): void;
+
+  /**
+   * Shows more field suggestions for the current query.
+   */
+  showMoreResults(): void;
+
+  /**
+   * Requests field suggestions based on a query.
+   */
+  search(): void;
+
+  /**
+   * If a facet exists with the configured `facetId`, selects a facet value.
+   *
+   * @param value - The field suggestion for which to select the matching facet value.
+   */
+  select(value: CategoryFieldSuggestionsValue): void;
+
+  /**
+   * Resets the query and empties the suggestions.
+   */
+  clear(): void;
+
+  /**
+   * Updates the captions of field suggestions.
+   *
+   * @param captions - A dictionary that maps field values to field suggestion display names.
+   */
+  updateCaptions(captions: Record<string, string>): void;
+
+  state: CategoryFieldSuggestionsState;
+}
 
-export interface CategoryFieldSuggestionsOptions extends CategoryFacetOptions {}
+// TODO: In v2, move these options into `FieldSuggestionsOptions` and remove `facetSearch`.
+export interface CategoryFieldSuggestionsFacetSearchOptions {
+  /**
+   * A dictionary that maps field values to field suggestion display names.
+   */
+  captions?: Record<string, string>;
+
+  /**
+   * The maximum number of suggestions to request.
+   *
+   * @defaultValue `10`
+   */
+  numberOfValues?: number;
+
+  /**
+   * The query with which to request field suggestions.
+   */
+  query?: string;
+}
+
+export interface CategoryFieldSuggestionsOptions {
+  /**
+   * The field for which you wish to get field suggestions.
+   */
+  field: string;
+
+  /**
+   * The base path shared by all values for the field suggestions.
+   *
+   * @defaultValue `[]`
+   */
+  basePath?: string[];
+
+  /**
+   * The character that specifies the hierarchical dependency.
+   *
+   * @defaultValue `;`
+   */
+  delimitingCharacter?: string;
+
+  /**
+   * A unique identifier for the controller. By default, a random unique identifier is generated.
+   *
+   * If a facet shares the same id, then its values are going to be selectable with `select` and `singleSelect`. However, you must make sure to build the field suggestion controller after the facet controller.
+   */
+  facetId?: string;
+
+  /**
+   * The options related to search.
+   */
+  facetSearch?: CategoryFieldSuggestionsFacetSearchOptions;
+
+  /**
+   * Whether to filter the results using `basePath`.
+   *
+   * @defaultValue `true`
+   */
+  filterByBasePath?: boolean;
+
+  /**
+   * Whether to exclude the parents of folded results when estimating the result count for each field suggestion.
+   *
+   * @defaultValue `true`
+   */
+  filterFacetCount?: boolean;
+
+  /**
+   * @deprecated This option has no effect.
+   */
+  injectionDepth?: number;
+
+  /**
+   * This option has no effect.
```

Comment:
I guess this isn't marked as deprecated because you need to wait until v2, right?

Reply:
it's because it will eventually have an effect once we flatten `facetSearch`.